variable_set.go

```diff
@@ -330,6 +379,47 @@ func (s *variableSets) RemoveFromWorkspaces(ctx context.Context, variableSetID s
 	return req.Do(ctx, nil)
 }
 
+// ************************
+// Apply variable set to projects in the supplied list.
+// Note: this method will return an error if the variable set has global = true.
+func (s *variableSets) ApplyToProjects(ctx context.Context, variableSetID string, options *VariableSetApplyToProjectsOptions) error {
+	if !validStringID(&variableSetID) {
+		return ErrInvalidVariableSetID
+	}
+	if err := options.valid(); err != nil {
+		return err
+	}
+
+	u := fmt.Sprintf("varsets/%s/relationships/projects", url.QueryEscape(variableSetID))
+	req, err := s.client.NewRequest("POST", u, options.Projects)
```

Comment:
Ah, thanks for pointing out that example @brandonc. I think the projects endpoint is a little different than VariableSetVariable Create because it expects an array of `data` (like [its workspaces sibling](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/variable-sets#apply-variable-set-to-workspaces)):

```
POST varsets/:varset_external_id/relationships/projects

{
  "data": [
    {
      "type": "projects",
      "id": "prj-YwfuBJZkdai4xj9w"
    }
  ]
}
```

versus VariableSetVarset Create, which wants a [single object](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/variable-sets#add-variable):

```
POST varsets/:varset_external_id/relationships/vars

{
  "data": {
    "type": "vars",
    "attributes": {
      "key": "g6e45ae7564a17e81ef62fd1c7fa86138",
      "value": "61e400d5ccffb3782f215344481e6c82",
      "description": "cheeeese",
      "sensitive": false,
      "category": "terraform",
      "hcl": false
    }
  }
}
```

I couldn't find a jsonapi decorator to turn that options struct into an array of objects under a root-level `data` key... does it make sense to change that method signature to take a slice of Projects instead, something like this?

```go
func (s *variableSets) ApplyToProjects(ctx context.Context, variableSetID string, projects []Project)
...
	u := fmt.Sprintf("varsets/%s/relationships/projects", url.QueryEscape(variableSetID))
	req, err := s.client.NewRequest("POST", u, projects)
```

Reply:
Ah, I didn't even realize this was an array. With a body like that, there it not a lot of room for other meaningful arguments. Here's a summary of our current conventions:

- AdminOrganizations UpdateModuleConsumers: accepts consumerOrganizationIDs []string and wraps each in a special purpose jsonapi ID/type struct
- OrganizationTags AddWorkspaces: accepts options struct with ProjectIDs []string and re-wraps them in the same type of special purpose jsonapi ID/type struct
- PolicySet AddPolicies: accepts options struct with a full []Policy member -- I think this is similar to what you are doing in this change

Having all that context, I think what is implemented is the best choice! Sorry for my incompetence.